src/morphology_workflows/repair.py

```diff
@@ -341,60 +344,72 @@ def _get_layer_mtype(mtype_input):
+            data["repair_morph_path"], repair_release_path
+        )
+    return name, data, m
+
+
 def make_release(df, _, zero_diameter_path, unravel_path, repair_path, extensions):
     """Make morphology release."""
     for extension in extensions:
+        _zero_diameter_path = None
         if zero_diameter_path is not None:
             _zero_diameter_path = Path(f"{zero_diameter_path}-{extension[1:]}")
             _zero_diameter_path.mkdir(exist_ok=True, parents=True)
 
+        _unravel_path = None
         if unravel_path is not None:
             _unravel_path = Path(f"{unravel_path}-{extension[1:]}")
             _unravel_path.mkdir(exist_ok=True, parents=True)
 
+        _repair_path = None
         if repair_path is not None:
             _repair_path = Path(f"{repair_path}-{extension[1:]}")
             _repair_path.mkdir(exist_ok=True, parents=True)
 
-        _m = []
-        for name in df.loc[df["is_valid"]].index:
-
-            mtype, layer = _get_layer_mtype(df.loc[name, "mtype"])
-            _m.append(
-                MorphInfo(
-                    name=name,
-                    mtype=mtype,
-                    layer=layer,
-                    use_dendrite=df.loc[name, "has_basal"],
-                    use_axon=df.loc[name, "has_axon"],
-                )
-            )
-
-            if _zero_diameter_path is not None:
-                zero_diameter_release_path = (
-                    str(_zero_diameter_path / Path(df.loc[name, "zero_diameter_morph_path"]).stem)
-                    + extension
-                )
-
-                df.loc[name, f"zero_diameter_release_morph_path_{extension[1:]}"] = _convert(
-                    df.loc[name, "zero_diameter_morph_path"], zero_diameter_release_path
-                )
-
-            if _unravel_path is not None:
-                unravel_release_path = (
-                    str(_unravel_path / Path(df.loc[name, "unravel_morph_path"]).stem) + extension
-                )
-                df.loc[name, f"unravel_release_morph_path_{extension[1:]}"] = _convert(
-                    df.loc[name, "unravel_morph_path"], unravel_release_path
-                )
+        __create_db_row = partial(
+            _create_db_row,
+            zero_diameter_path=_zero_diameter_path,
+            unravel_path=_unravel_path,
+            repair_path=_repair_path,
+            extension=extension,
+        )
 
-            if _repair_path is not None:
-                repair_release_path = (
-                    str(_repair_path / Path(df.loc[name, "repair_morph_path"]).stem) + extension
-                )
-                df.loc[name, f"repair_release_morph_path_{extension[1:]}"] = _convert(
-                    df.loc[name, "repair_morph_path"], repair_release_path
-                )
+        _m = []
+        with Pool() as pool:
+            for name, row, m in pool.imap(__create_db_row, df.loc[df["is_valid"]].iterrows()):
```

Comment:
Parallelization is different from vectorization :sweat_smile: But that's nice too.
Anyway, `pd.DataFrame.iterrows()` is more or less the slowest way to iterate over a DF (`itertuples` is usually a bit faster but still very slow compared to other methods), so it should almost never be used (see https://stackoverflow.com/a/55557758). Here I think `df.loc[df["is_valid"]].to_dict("index")` would be a bit better. But the best would be to split the DF into several parts, use one part in each process and use `apply` on each of these parts (that's what we do in `BluePyParallel`). But I think that's a bit overkill here.

Reply:
I know haha, but the slow part here is _convert by far. So this does the trick. We can wait 5min to have a release of a few hundred morphologies, just time for a coffee

Reply:
ahah ok :smile: